W/Wayland/build_tarballs.jl

```diff
@@ -1,32 +1,47 @@
 # Note that this script can accept some limited command-line arguments, run
 # `julia build_tarballs.jl --help` to see a usage message.
-using BinaryBuilder
+using BinaryBuilder, Pkg
 
 name = "Wayland"
-version = v"1.21.0"
+version = v"1.23.1"
 
 # Collection of sources required to build Wayland
 sources = [
     ArchiveSource("https://gitlab.freedesktop.org/wayland/wayland/-/releases/$(version)/downloads/wayland-$(version).tar.xz",
-                  "6dc64d7fc16837a693a51cfdb2e568db538bfdc9f457d4656285bb9594ef11ac"),
+                  "864fb2a8399e2d0ec39d56e9d9b753c093775beadc6022ce81f441929a81e5ed"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/wayland-*/
 
-ln -s `which wayland-scanner` $bindir
-cp $prefix/libdata/pkgconfig/* $prefix/lib/pkgconfig || true
+mkdir build-host
+cd build-host
+
+# Hack to make the pkgconfig paths work on the host build
```

Comment:
@giordano I want to draw your attention to 2 hacks in here to build the host-tool wayland-scanner on every platform, to work around these issues:
- Host pkgconfig prefixes all paths with a superfluous `/workspace/destdir`
- Target libraries end up in the host sysroot, at least on freeBSD
Possibly these are bugs somewhere in the sysimages.

Reply:
Open an issue, I won't have the time to look into this for a while.